Dbz 8281 Provide outbox Quarkus devservice

dcdh opened the pull request.

dcdh committed:
DBZ-8281 align pom following a 3.15.1 extension version

dcdh committed:
DBZ-8281 update Aliases.txt and COPYRIGHT.txt

dcdh committed:
DBZ-8281 use devservice instead of custom DatabaseTestResource and aligned integration tests between blocking and reactive test modules

dcdh committed:
DBZ-8281 implement outbox devservice

dcdh commented on debezium-quarkus-outbox-common/deployment/src/main/java/io/quarkus/kafka/client/deployment/RedpandaKafkaContainer.java:
```diff
@@ -0,0 +1,135 @@
+/*
```
hack, will remove it when changed will be pushed and release inside Quarkus
Keep pr in draft until done

dcdh replied:
https://github.com/quarkusio/quarkus/pull/43582

dcdh committed:
DBZ-8281 use test scope to avoid test dependencies inside production code

Naros commented on debezium-quarkus-test-outbox/src/main/java/io/debezium/outbox/test/quarkus/remote/DebeziumConnectorStatus.java:
```diff
@@ -0,0 +1,25 @@
+
+import java.util.Objects;
+
+public record DebeziumConnectorStatus(String name, Connector connector) {
```
Could use some documentation.

Naros commented on debezium-quarkus-test-outbox/src/main/java/io/debezium/outbox/test/quarkus/remote/RemoteConnectorApi.java:
```diff
@@ -0,0 +1,51 @@
+import io.quarkus.rest.client.reactive.QuarkusRestClientBuilder;
+
+@Path("/")
+public interface RemoteConnectorApi {
```
Can we document the purpose and use of this API.

Naros reviewed (changes requested):
I've left a few inline comments @dcdh, if you could take a look.
If you could also rebase the PR, we'd appreciate it.

dcdh commented on debezium-quarkus-outbox-reactive/integration-tests/src/main/resources/application.properties:
```diff
@@ -1,18 +1,25 @@
 #
 #quarkus.debezium-outbox.payload.column-definition=JSONB
-#quarkus.debezium-outbox.payload.type=com.vladmihalcea.hibernate.type.json.JsonNodeBinaryType
+#quarkus.debezium-outbox.payload.type=com.vladmihalcea.hikbernate.type.json.JsonNodeBinaryType
```
Yes it is a typo. I did not see it.
Thanks for reporting it

dcdh committed:
DBZ-8281 document Debezium connector rest api

dcdh commented:
@Naros 

Thanks for reviewing this PR. I have resolved all of your comments.

Inside `DebeziumConnectorStatus` I have removed `state()` method because it is not used.

I forget to do little commits for the review instead I rewrite them. I am sorry...

Documentation has been generated using chatgpt, lot easier way.

Regards,
Damien

dcdh committed:
DBZ-8281 rework is running state encapsulation

dcdh committed:
DBZ-8281 refactor: extract JdbcDataSourceSourceConnector into separate class

- Moved JdbcDataSourceSourceConnector into its own class to improve modularity and reusability.
- This extraction helps isolate database connector logic and simplifies future extensions.

dcdh committed:
DBZ-8281 Return the databaseServerName for topic.prefix, used exclusively for logging messages. This ensures uniqueness and helps maintain clear and organized logging.

Naros commented:
:x: Developer Certificate of Origin (DCO) check failed.

Hi @dcdh, please sign off all commits with:

```
git commit -s
```

If pull request commits are not signed off, the pull request cannot be merged. For more information about why this is required, please see [our blog](https://debezium.io/blog/2025/12/12/contribution-requirements-changing/) about contribution requirement changes.

Naros commented:
Thank you for your contribution.

This pull request is being closed due to extended inactivity. If you would like to continue working on this, please feel free to reopen this pull request or submit a new one against the latest main branch. We appreciate your interest in improving Debezium.
